apps/studio/src/lib/db/clients/BasicDatabaseClient.ts

```diff
@@ -30,42 +34,78 @@ export interface AppContextProvider {
 }
 
 // raw result type is specific to each database implementation
-export abstract class BasicDatabaseClient<RawResultType> implements DatabaseClient {
+export abstract class BasicDatabaseClient<RawResultType> {
 
   knex: Knex | null;
   contextProvider: AppContextProvider
+  server: IDbConnectionServer;
+  database: IDbConnectionDatabase;
+  connectionType: ConnectionType;
 
-
-  constructor(knex: Knex | null, contextProvider: AppContextProvider) {
+  constructor(knex: Knex | null, contextProvider: AppContextProvider, server: IDbConnectionServer, database: IDbConnectionDatabase) {
     this.knex = knex;
     this.contextProvider = contextProvider
-  }
-  listTablePartitions(_table: string, _schema: string): Promise<TablePartition[]> {
-    return Promise.resolve([])
-  }
-  alterPartitionSql(_changes: AlterPartitionsSpec): string {
-    return ''
+    this.server = server;
+    this.database = database;
```

Comment:
How do we get rid of this `server` and `database` nonsense? Like what is the `database`?